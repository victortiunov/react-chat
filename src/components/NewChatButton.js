import React from 'react';
import { withStyles } from 'material-ui/styles';
import Button from 'material-ui/Button';
import AddIcon from 'material-ui-icons/Add';
import Modal from 'material-ui/Modal';
import Typography from 'material-ui/Typography';
import TextField from 'material-ui/TextField';

const styles = theme => ({
  newChatButton: {
    position: 'absolute',
    left: 'auto',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3 + 48,
  },
  modalDialog: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalForm: {
    width: '30%',
    minWidth: '300px',
    padding: theme.spacing.unit * 3,
    backgroundColor: theme.palette.background.paper,
  },
});

class NewChatButton extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      modalView: false,
      chatTitle: '',
    };
  }

  toggleModal = () => {
    this.setState({ modalView: !this.state.modalView });
  };

  handleTitleChange = (e) => {
    this.setState({ chatTitle: e.target.value });
  };

  handleSubmit = (e) => {
    e.preventDefault();

    this.props.createChat(this.state.chatTitle.trim());
    this.toggleModal();
    this.setState({ chatTitle: '' });
  };

  render() {
    const { classes, disabled } = this.props;
    const { modalView, chatTitle } = this.state;

    return (
      <React.Fragment>
        <Button
          variant="fab"
          color="primary"
          className={classes.newChatButton}
          onClick={this.toggleModal}
          disabled={disabled}
        >
          <AddIcon />
        </Button>
        <Modal
          className={classes.modalDialog}
          open={modalView && !disabled}
          onClose={this.toggleModal}
        >
          <form className={classes.modalForm} onSubmit={this.handleSubmit}>
            <Typography variant="title" id="modal-title">
              Create new chat
            </Typography>
            <TextField
              required
              fullWidth
              autoFocus
              label="New chat title"
              placeholder="Type here the title..."
              type="text"
              margin="normal"
              autoComplete="new-chat"
              value={chatTitle}
              onChange={this.handleTitleChange}
            />
            <Button variant="raised" type="submit" color="primary" disabled={!chatTitle.trim()}>
              Create
            </Button>
          </form>
        </Modal>
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(NewChatButton);
