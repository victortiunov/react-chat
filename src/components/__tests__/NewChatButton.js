/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NewChatButton from '../NewChatButton';

const mockProps = {
  createChat: jest.fn(),
  disabled: false,
};

describe('<NewChatButton />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NewChatButton {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders button', () => {
    const tree = renderer.create(<NewChatButton {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled button', () => {
    const tree = renderer.create(<NewChatButton {...mockProps} disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
