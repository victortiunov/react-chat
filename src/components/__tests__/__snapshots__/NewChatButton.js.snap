// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewChatButton /> renders button 1`] = `
<button
  className="MuiButtonBase-root-55 MuiButton-root-40 MuiButton-raised-45 MuiButton-fab-50 MuiButton-raisedPrimary-47 NewChatButton-newChatButton-37"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-41"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-58"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-64"
  />
</button>
`;

exports[`<NewChatButton /> renders disabled button 1`] = `
<button
  className="MuiButtonBase-root-55 MuiButtonBase-disabled-56 MuiButton-root-40 MuiButton-raised-45 MuiButton-fab-50 MuiButton-raisedPrimary-47 MuiButton-disabled-49 NewChatButton-newChatButton-37"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="-1"
  type="button"
>
  <span
    className="MuiButton-label-41"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-58"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </span>
</button>
`;
